repair/row_level.cc

```diff
@@ -1085,23 +1085,45 @@ class repair_meta {
         });
     }
 
+    future<std::list<repair_row>>
+    copy_rows_from_working_row_buf() {
+        return do_with(std::list<repair_row>(), [this] (std::list<repair_row>& rows) {
+            return do_for_each(_working_row_buf, [this, &rows] (const repair_row& r) {
+                rows.push_back(r);
+            }).then([&rows] {
+                return std::move(rows);
+            });
+        });
+    }
+
+    future<std::list<repair_row>>
+    copy_rows_from_working_row_buf_within_set_diff(std::unordered_set<repair_hash> set_diff) {
+        return do_with(std::list<repair_row>(), std::move(set_diff),
+                [this] (std::list<repair_row>& rows, std::unordered_set<repair_hash>& set_diff) {
+            return do_for_each(_working_row_buf, [this, &set_diff, &rows] (const repair_row& r) {
+                if (set_diff.count(r.hash()) > 0) {
+                    rows.push_back(r);
+                }
+            }).then([&rows] {
+                return std::move(rows);
+            });
+        });
+    }
+
     // Return rows in the _working_row_buf with hash within the given sef_diff
     // Give a set of row hashes, return the corresponding rows
     // If needs_all_rows is set, return all the rows in _working_row_buf, ignore the set_diff
-    std::list<repair_row>
-    get_row_diff(const std::unordered_set<repair_hash>& set_diff, needs_all_rows_t needs_all_rows = needs_all_rows_t::no) {
-        std::list<repair_row> rows;
+    future<std::list<repair_row>>
+    get_row_diff(std::unordered_set<repair_hash> set_diff, needs_all_rows_t needs_all_rows = needs_all_rows_t::no) {
         if (needs_all_rows) {
+            std::list<repair_row> rows;
             if (!_repair_master || _nr_peer_nodes == 1) {
```

Comment:
unused?

Reply:
Yes, I will remove it. It is not used after switching to the helpers.